classes/client/s3_client.php

```diff
@@ -54,6 +56,72 @@ public function set_client($config) {
         ));
     }
 
+    /**
+     * @return mixed
+     */
+    public function define_settings_form() {
+        $connectiontest = self::test_connection();
```

Comment:
Is there a particular reason why this uses `self::` instead of `$this->` seeing as the function itself is not static?